### DISCUSSION TITLE
Why is Boolean not true and false?

### DISCUSSION THREAD

**Discussion by jaulz** [millsp/ts-toolbelt#303] [Q&A]
First of all, thanks for this cool library! However, I wonder why you used `1` and `0` for Boolean instead of `true` and `false`? Something like this causes an issue right now because the return parameter is not a real boolean:
```typescript
TPartial extends boolean = Contains<TParams, typeof Wildcard>
```

Did I miss anything?

**Comment by sashulinator** [comment]
Hi!
I guess because with Boolean you can't do something like that
```ts
export type Select<U extends any, M extends any, match extends Match = 'default'> =
    U extends unknown
    ? {1: U & M, 0: never}[Is<U, M, match>] // HERE IS THE DEAL
    : never
```
